File: ProyectoMysql/server/BusinessLayer/OrdenCompra.py
from DataLayer.OrdenCompraDB import *
from EntityLayer.OrdenCompraEntity import *
from Utilidades.Conexion.configMysql import StartTransaction, EndTransaction, Restore

class OrdenCompra:

    def ObtenerMain():
        try:
            return OrdenCompraDB.ObtenerMain()
        except Exception as e:
            print(e)

    def ObtenerItem(EmpresaId: int):
        try:
            return OrdenCompraDB.ObtenerItem(EmpresaId)
        except Exception as e:
            print(e)

    def Registrar(Ent: OrdenCompraSaveModel):
        try:    
            StartTransaction()
            data= OrdenCompraDB.Registrar(Ent)
            EndTransaction()
            return data.OrdenCompraId
        except Exception as e:
            print(e)
    

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.BusinessLayer/OrdenCompra.cs
using LogisticStorage.DataLayer;
using LogisticStorage.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.BusinessLayer
{
    public class OrdenCompra
    {
        public static List<OrdenCompraEntity> ObtenerMain()
        {
            OrdenCompraDB DB = new OrdenCompraDB();
            return DB.ObtenerMain();
        }
        public static List<OrdenCompraEntity> ObtenerItem(Int32 EmpresaId)
        {
            OrdenCompraDB DB = new OrdenCompraDB();
            return DB.ObtenerItem(EmpresaId);
        }
        public static Int32 Registrar(OrdenCompraEntity Ent)
        {
            OrdenCompraDB DB = new OrdenCompraDB();
            DB.Registrar(Ent);

            return Ent.OrdenCompraId;
        }
    }
}
